examples/gc.ml

```diff
@@ -0,0 +1,99 @@
+open Lwt.Syntax
+
+(** make schema *)
+
+module Schema = Irmin.Schema.KV (Irmin.Contents.String)
+
+(** define configuration parameters for pack store *)
+
+module Store_conf : Irmin_pack.Conf.S = struct
+  (** number of entries in the internal inode B-tree. 32 performs well *)
+  let entries = 32
+
+  (** set this to 0 to only use inode hashing (this paraemter exists for
+      particular backwards compatibility applications with Irmin 1.0 data) *)
+  let stable_hash = 0
+
+  (** GC only supports minimal indexing strategy so [contents_length_header]
+      must not be [None], which requires an always indexing strategy *)
+  let contents_length_header = Some `Varint
+
+  (** this is the recommended sorting algorithm for new stores *)
+  let inode_child_order = `Hash_bits
+
+  (** throw an error (or not) if an empty tree is persisted *)
+  let forbid_empty_dir_persistence = true
+end
+
+(** make pack store *)
+
+module Maker = Irmin_pack_unix.Maker (Store_conf)
+module Store = Maker.Make (Schema)
+
+(** define configuration for repo *)
+
+module Repo_config : sig
+  val config : Irmin.config
+end = struct
+  (** how to handle large writes *)
+  let merge_throttle = `Block_writes
+
+  (** must use minimal indexing strategy to use GC *)
+  let indexing_strategy = Irmin_pack.Indexing_strategy.minimal
+
+  (** setting to true trades off performance for durability *)
+  let use_fsync = false
+
+  (** buffer size that triggers auto flushing to disk *)
+  let dict_auto_flush_threshold = 1_000_000
+
+  (** buffer size that triggers auto flushing to disk *)
+  let suffix_auto_flush_threshold = 1_000_000
```

Comment:
Any guidance on how to configure these that would be useful for an end-user? I copied over the default values here.

Also, did we consider having one value for auto-flushing that applies for both files? That seems to me like it would be a better configuration option to expose for end-users.

Reply:
I think it is better to keep the dict and the suffix auto_flush_thresholds separate - they are separate components and the way they are used is quite different (no reason that their thresholds should be the same).

Reply:
:+1: It sounds like we might need to think about what our default values are for these two thresholds. Do you agree?

Reply:
https://github.com/mirage/irmin/issues/2018